crate/utils/src/crypto/hybrid_encryption/rsa_oaep_aes_gcm.rs

```diff
@@ -0,0 +1,174 @@
+use cloudproof::reexport::crypto_core::reexport::zeroize::Zeroizing;
+use openssl::{
+    pkey::{PKey, Private, Public},
+    rand::rand_bytes,
+    rsa::Padding,
+    symm::{decrypt_aead, encrypt_aead, Cipher},
+};
+
+use crate::{
+    crypto::symmetric::{AES_256_GCM_IV_LENGTH, AES_256_GCM_KEY_LENGTH, AES_256_GCM_MAC_LENGTH},
+    error::KmipUtilsError,
+    kmip_utils_bail,
+};
+
+#[cfg(feature = "fips")]
+const FIPS_MIN_RSA_MODULUS_LENGTH: u32 = 256;
+
+/// Asymmetrically encrypt data referring to PKCS#11 available at
+/// http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cos01/pkcs11-curr-v2.40-cos01.html
+///
+/// Let `m` be the message to encrypt, first generate a temporary random AES key
+/// `kek`. Encrypt it using RSA-OAEP; `c` is the encrypted key.
+///
+/// Encrypt the message `m` such as`ct = enc(kek, m)` using the key `kek`
+/// with AES-256-GCM with proper IV and potential additional data.
+///
+/// Send `c|iv|ct|tag` where `|` is the concatenation operator, `iv` the
+/// initialization vector and `tag` the authentication tag.
+///
+/// TODO - support OAEP for different hashes.
+pub fn rsa_oaep_aes_gcm_encrypt(
+    pubkey: PKey<Public>,
+    plaintext: &[u8],
+    aad: Option<&[u8]>,
+) -> Result<Vec<u8>, KmipUtilsError> {
+    let rsa_pubkey = pubkey.rsa()?;
+    #[cfg(feature = "fips")]
+    if rsa_pubkey.size() < FIPS_MIN_RSA_MODULUS_LENGTH {
+        kmip_utils_bail!(
+            "CKM_RSA_OAEP encryption error: RSA key has insufficient size: expected >= {} bytes \
+             and got {} bytes",
+            FIPS_MIN_RSA_MODULUS_LENGTH,
+            pubkey.bits()
+        )
+    }
+
+    let encapsulation_bytes_len = rsa_pubkey.size() as usize;
+
+    // Generate temporary AES key.
+    let mut kek = Zeroizing::from(vec![0u8; AES_256_GCM_KEY_LENGTH]);
+    rand_bytes(&mut kek)?;
```

Comment:
Just a general comment: at one point, we should not depend on crypto_core anymore (except for Findex): since Openssl do not provide zeroize implementation, IMHO we should use directly zeroize.

Reply:
So instead of importing `cryptocore::reexport::zeroize` we better import `zeroize` directly ?
Maybe consider doing this in another issue ?